balancer/base/balancer.go

```diff
@@ -200,14 +224,21 @@ func (b *baseBalancer) UpdateSubConnState(sc balancer.SubConn, state balancer.Su
 	oldAggrState := b.state
 	b.state = b.csEvltr.RecordTransition(oldS, s)
 
+	// Set or clear the last connection error accordingly.
+	if b.state == connectivity.TransientFailure {
+		b.connErr = fmt.Errorf("last connection error %v", state.ConnectionError)
```

Comment:
If we do the string "last connection error " concatenation when generating picker, can this be just `b.connErr = state.ConnectionError`?

Reply:
Hmm, yes that makes sense.  Done.